homeassistant/components/fan/xiaomi_miio.py

```diff
@@ -0,0 +1,354 @@
+                                          SUPPORT_SET_SPEED, )
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (CONF_NAME, CONF_HOST, CONF_TOKEN,
+                                 ATTR_ENTITY_ID, )
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Xiaomi Air Purifier'
+PLATFORM = 'xiaomi_miio'
+DOMAIN = 'airpurifier'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): vol.All(cv.string, vol.Length(min=32, max=32)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+REQUIREMENTS = ['python-mirobo==0.2.0']
+
+ATTR_TEMPERATURE = 'temperature'
+ATTR_HUMIDITY = 'humidity'
+ATTR_AIR_QUALITY_INDEX = 'aqi'
+ATTR_MODE = 'mode'
+ATTR_FILTER_HOURS_USED = 'filter_hours_used'
+ATTR_FILTER_LIFE = 'filter_life_remaining'
+ATTR_FAVORITE_LEVEL = 'favorite_level'
+ATTR_BUZZER = 'buzzer'
+ATTR_CHILD_LOCK = 'child_lock'
+ATTR_LED = 'led'
+ATTR_LED_BRIGHTNESS = 'led_brightness'
+ATTR_MOTOR_SPEED = 'motor_speed'
+
+ATTR_BRIGHTNESS = 'brightness'
+ATTR_LEVEL = 'level'
+
+SUCCESS = ['ok']
+
+SERVICE_SET_BUZZER_ON = 'set_buzzer_on'
+SERVICE_SET_BUZZER_OFF = 'set_buzzer_off'
+SERVICE_SET_LED_ON = 'set_led_on'
+SERVICE_SET_LED_OFF = 'set_led_off'
+SERVICE_SET_FAVORITE_LEVEL = 'set_favorite_level'
+SERVICE_SET_LED_BRIGHTNESS = 'set_led_brightness'
+
+AIRPURIFIER_SERVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+SERVICE_SCHEMA_LED_BRIGHTNESS = AIRPURIFIER_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_BRIGHTNESS):
+        vol.All(vol.Coerce(int), vol.Clamp(min=0, max=2))
+})
+
+SERVICE_SCHEMA_FAVORITE_LEVEL = AIRPURIFIER_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_LEVEL):
+        vol.All(vol.Coerce(int), vol.Clamp(min=0, max=17))
+})
+
+SERVICE_TO_METHOD = {
+    SERVICE_SET_BUZZER_ON: {'method': 'async_set_buzzer_on'},
+    SERVICE_SET_BUZZER_OFF: {'method': 'async_set_buzzer_off'},
+    SERVICE_SET_LED_ON: {'method': 'async_set_led_on'},
+    SERVICE_SET_LED_OFF: {'method': 'async_set_led_off'},
+    SERVICE_SET_FAVORITE_LEVEL: {
+        'method': 'async_set_favorite_level',
+        'schema': SERVICE_SCHEMA_FAVORITE_LEVEL},
+    SERVICE_SET_LED_BRIGHTNESS: {
+        'method': 'async_set_led_brightness',
+        'schema': SERVICE_SCHEMA_LED_BRIGHTNESS},
+}
+
+
+# pylint: disable=unused-argument
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the air purifier from config."""
+    from mirobo import AirPurifier, DeviceException
+    if PLATFORM not in hass.data:
+        hass.data[PLATFORM] = {}
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    token = config.get(CONF_TOKEN)
+
+    _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
```

Comment:
.debug() here?

Reply:
It's in sync with some other miio components. I prefer the log level info.

Reply:
Fair enough.

Reply:
We should just remove them from all components so we are in sync with the rest of Home Assistant.